From 37034f930913dc6b58634b9946d924f2249cd7ba Mon Sep 17 00:00:00 2001
From: Samuli Piippo <samuli.piippo@qt.io>
Date: Mon, 20 Jan 2020 11:17:57 +0200
Subject: [PATCH] Avoid conflicting declaration for egl typedefs

The EGL typedefs may have alternative declaration on some platforms,
such as NVIDIA, which cause build errors. Add guards around the
typedefs to avoid re-declaration.

Task-number: QTBUG-79709
Change-Id: I14f09c94e159ef51dd858f0877fe610ce6f582d4
---
 .../compositor/linux-dmabuf-unstable-v1/linuxdmabuf.h           | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/hardwareintegration/compositor/linux-dmabuf-unstable-v1/linuxdmabuf.h b/src/hardwareintegration/compositor/linux-dmabuf-unstable-v1/linuxdmabuf.h
index 02b5b6f8..56a710c3 100644
--- a/src/hardwareintegration/compositor/linux-dmabuf-unstable-v1/linuxdmabuf.h
+++ b/src/hardwareintegration/compositor/linux-dmabuf-unstable-v1/linuxdmabuf.h
@@ -59,8 +59,10 @@
 #endif
 
 // Copied from eglmesaext.h
+#ifndef EGL_WL_bind_wayland_display
 typedef EGLBoolean (EGLAPIENTRYP PFNEGLBINDWAYLANDDISPLAYWL) (EGLDisplay dpy, struct wl_display *display);
 typedef EGLBoolean (EGLAPIENTRYP PFNEGLUNBINDWAYLANDDISPLAYWL) (EGLDisplay dpy, struct wl_display *display);
+#endif
 
 QT_BEGIN_NAMESPACE
 
